diff --git a/sunriset.cpp b/sunriset.cpp
index ed6ee0b..f7f7974 100755
--- a/sunriset.cpp
+++ b/sunriset.cpp
@@ -148,7 +148,7 @@ void sunpos (const double d, double *lon, double *r)
 void sun_RA_dec (const double d, double *RA, double *dec, double *r)
 {
   double lon, obl_ecl;
-  double xs, ys, zs;
+  double xs, ys;
   double xe, ye, ze;
   
   /* Compute Sun's ecliptical coordinates */
@@ -157,7 +157,6 @@ void sun_RA_dec (const double d, double *RA, double *dec, double *r)
   /* Compute ecliptic rectangular coordinates */
   xs = *r * cosd(lon);
   ys = *r * sind(lon);
-  zs = 0; /* because the Sun is always in the ecliptic plane! */
 
   /* Compute obliquity of ecliptic (inclination of Earth's axis) */
   obl_ecl = 23.4393 - 3.563E-7 * d;
diff --git a/sunwait.cpp b/sunwait.cpp
index 7e43bc9..8ff6429 100755
--- a/sunwait.cpp
+++ b/sunwait.cpp
@@ -662,7 +662,7 @@ int main (int argc, char *argv[])
     if (pRun->debug == ONOFF_ON) printf ("Debug: argv[%d]: >%s<\n", i, arg);
 
     // Strip any hyphen from arguments, but not negative signs of numbers
-    if (arg[0] == '-' && arg[1] != '\0' && !isdigit(arg[1])) *arg++;
+    if (arg[0] == '-' && arg[1] != '\0' && !isdigit(arg[1])) memmove(arg, arg + 1, sizeof arg - 1);
 
     // Normal help or version info
          if   (!strcmp (arg, "v")             ||
